src/iterators/SingleCellIteratorInterface.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * SingleCellIteratorInterface.h
+ *
+ *  Created on: 31 May 2018
+ *      Author: seckler
+ */
+
+#pragma once
+#include "ParticleIteratorInterface.h"
+
+namespace autopas {
+
+/**
+ * SingleCellIteratorInterface class to loop over particles of a single cell.
+ *
+ * @tparam Particle type of the Particles
+ */
+template <class Particle>
+class SingleCellIteratorInterface : public ParticleIteratorInterface<Particle> {
+ public:
+  /**
+   * default constructor of SingleCellIteratorInterface
+   */
+  SingleCellIteratorInterface() {}
+
+  /**
+   * destructor of SingleCellIteratorInterface
+   */
+  virtual ~SingleCellIteratorInterface() = default;
+
+  /**
+   * equality operator.
+   * if both iterators are invalid or if they point to the same particle, this
+   * returns true
+   * @param rhs
+   * @return
+   */
+  virtual bool operator==(const SingleCellIteratorInterface<Particle> &rhs) const = 0;
+
+  /**
+   * inequality operator
+   * descrition see operator==
+   * @param rhs
+   * @return
+   */
+  virtual bool operator!=(const SingleCellIteratorInterface<Particle> &rhs) const = 0;
+
+  /**
+   * Get the index of the particle in the cell
+   * @return index of the current particle
+   */
+  virtual int getIndex() const = 0;
+};
+
+namespace internal {
+
+/**
+ * All implementations of the interface should inherit from this class. It extends the interface just by the clone
+ * method, which is needed by the Wrapper.
+ * @tparam Particle
+ */
+template <class Particle>
+class SingleCellIteratorInterfaceImpl : public SingleCellIteratorInterface<Particle> {
```

Comment:
Naive question: Why do we need this? Can the `clone()` method not go into the `SingleCellIteratorInterface`? And if not please explain in the doc the why.

Reply:
the reason:
the wrapper should not have that method. It is also completely undefined how the wrapper would handle that. I also don't want it to be visible for the wrapper to a user.